lib/qmp.ml

```diff
@@ -130,14 +141,24 @@ let message_of_string x =
       | `Assoc [] -> Unit
       | `Assoc list when List.mem_assoc "status" list ->
         Status (string (List.assoc "status" list))
-      | `Assoc list when List.mem_assoc "enabled" list ->
+      | `Assoc list when List.mem_assoc "enabled" list && List.mem_assoc "present" list ->
```

Comment:
I thought the lookup is sequential, moving the block at line 152 above this should make it get precedence and avoid this change. Am I wrong?